api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/AllergyIntoleranceTranslatorImpl.java

```diff
@@ -83,6 +106,11 @@ public Allergy toOpenmrsType(Allergy allergy, AllergyIntolerance fhirAllergy) {
 			}
 		}
 		allergy.setVoided(isAllergyInactive(fhirAllergy.getClinicalStatus()));
+		allergy.setPatient(
+		    patientTranslator.toOpenmrsType(patientService.getPatientByUuid(getReferenceId(fhirAllergy.getPatient()))));
```

Comment:
`allergy.setPatient(patientReferenceTranslator.toOpenmrsType(fhirAllergy.getPatient()))`

Reply:
Of course, you have to inject patientReferenceTranslator
```
@Inject
private PatientReferenceTranslator patientReferenceTranslator;
```